src/coreclr/jit/alloc.cpp

```diff
@@ -327,3 +327,47 @@ void ArenaAllocator::dumpMaxMemStats(FILE* file)
     s_maxStats.Print(file);
 }
 #endif // MEASURE_MEM_ALLOC
+
+void* operator new(std::size_t size)
+{
+    assert(!"heap new called; use malloc directly if heap allocation was intended");
+    if (size == 0)
```

Comment:
just curious, why don't we just fail here unconditionally? Do you suspect there might be more usages?

Reply:
Yes, for backporting reasons I was worried about turning some harmless/tiny memory leak into an unconditional fail fast.